src/nestegg.c

```diff
@@ -3004,6 +3043,26 @@ nestegg_packet_encryption(nestegg_packet * pkt)
 }
 
 int
+nestegg_packet_partitioned(nestegg_packet * pkt)
+{
+  struct frame * f = pkt->frame;
+  unsigned char partitioned_bit;
+
+  if (!f->frame_encryption)
+    return NESTEGG_PACKET_HAS_SIGNAL_BYTE_FALSE;
+
+  /* Should never have parsed blocks with both encryption and lacing */
+  assert(f->next == NULL);
+
+  partitioned_bit = f->frame_encryption->signal_byte & PARTITIONED_BIT_MASK;
```

Comment:
If we keep this instead of rolling it in to `nestegg_packet_encryption`, we should check that the encrypted bit is set and return an error if not, since the spec says encrypted must be set if partitioned is set.